project/src/main/puzzle/milestone-hud.gd

```diff
@@ -120,5 +122,26 @@ func _on_PuzzleScore_game_prepared() -> void:
 	init_milebar()
 
 
+"""
+We emit particles when the player levels up.
+
+This provides visual feedback for people playing without sound.
+"""
+func _on_PuzzleScore_speed_index_changed(value: int) -> void:
+	if value == 0:
+		# initializing the level; don't emit any particles
+		return
+	
+	var particles_material: ParticlesMaterial = _progress_bar_particles.get_child(0).process_material
```

Comment:
It's a little strange that we only modify one of the children; worth documenting

Reply:
Added comment, `All of the Particles2D share the same GradientTexture so we only need to modify one.`